cmd/notation/sign.go

```diff
@@ -112,8 +115,17 @@ func runSign(command *cobra.Command, cmdOpts *signOpts) error {
 	_, err = notation.Sign(ctx, signer, sigRepo, opts)
 	if err != nil {
 		var errorPushSignatureFailed notation.ErrorPushSignatureFailed
-		if errors.As(err, &errorPushSignatureFailed) && !ociImageManifest {
-			return fmt.Errorf("%v. Possible reason: target registry does not support OCI artifact manifest. Try removing the flag `--signature-manifest artifact` to store signatures using OCI image manifest", err)
+		if errors.As(err, &errorPushSignatureFailed) {
+			if !ociImageManifest {
+				return fmt.Errorf("%v. Possible reason: target registry does not support OCI artifact manifest. Try removing the flag `--signature-manifest artifact` to store signatures using OCI image manifest", err)
+			}
+			if _, after, found := strings.Cut(err.Error(), referrersTagSchemaDeleteError); found {
+				warnMsg := fmt.Sprintf("Warning: %s %s", referrersTagSchemaDeleteError, after)
```

Comment:
Why not using `err.Error()` directly here?
```suggestion
				warnMsg := fmt.Sprintf("Warning: signature successfully pushed with minor cleanup failure: %s", err.Error())
```

P.S. I also met issue on how to describe this error to customer without too much low-level detail for oras' fix.

Reply:
Because the full content of `err.Error()` starts with:  `Error: failed to push signature to registry with error: failed to push manifest:...`. However, in this case, it's not an error, both push signature and push manifest were succeeded. 
Note: `failed to push manifest` is an error message from oras-go: https://github.com/oras-project/oras-go/blob/e8225cb1e125bd4c13d6b586ae6d862050c3fae2/pack.go#L174.

Reply:
How about something like "Warning: signature is successfully pushed but the old referrer index failed to be cleaned up"?